### ISSUE TITLE
Error in extracting chromatogram object

### ISSUE THREAD

**Issue by marcomtzmtz** [sneumann/xcms#301]
Dear xcms devel

I get the following error when I try to exctract the chromatogram:

Thank you so much

bpis <- chromatogram(raw_data, aggregationFun = "max")
Error: BiocParallel errors
  element index: 1, 2, 3, 4, 5, 6, ...
  first error: missing value where TRUE/FALSE needed

### my traceback:

> traceback()
11: stop(.error_bplist(res))
10: bplapply(X = seq_along(ddd[[1L]]), wrap, .FUN = FUN, .ddd = ddd, 
        .MoreArgs = MoreArgs, BPREDO = BPREDO, BPPARAM = BPPARAM)
9: bplapply(X = seq_along(ddd[[1L]]), wrap, .FUN = FUN, .ddd = ddd, 
       .MoreArgs = MoreArgs, BPREDO = BPREDO, BPPARAM = BPPARAM)
8: bpmapply(subs_by_file, match(fileNames(subs), fns), FUN = function(cur_sample, 
       cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE)
7: bpmapply(subs_by_file, match(fileNames(subs), fns), FUN = function(cur_sample, 
       cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE)
6: withCallingHandlers(expr, warning = function(w) invokeRestart("muffleWarning"))
5: suppressWarnings(res <- bpmapply(subs_by_file, match(fileNames(subs), 
       fns), FUN = function(cur_sample, cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE))
4: .extractMultipleChromatograms(object, rt = rt, mz = mz, aggregationFun = aggregationFun, 
       missingValue = missing, msLevel = msLevel, BPPARAM = BPPARAM)
3: .local(object, ...)
2: chromatogram(raw_data, aggregationFun = "max")
1: chromatogram(raw_data, aggregationFun = "max")

### Session info

> sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.5

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_1.5        pander_0.6.2        RColorBrewer_1.1-2  faahKO_1.20.0      
 [5] xcms_3.2.0          MSnbase_2.6.1       ProtGenerics_1.12.0 mzR_2.14.0         
 [9] Rcpp_0.12.17        BiocParallel_1.14.1 Biobase_2.40.0      BiocGenerics_0.26.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4       purrr_0.2.5            splines_3.5.0         
 [4] lattice_0.20-35        colorspace_1.3-2       stats4_3.5.0          
 [7] yaml_2.1.19            vsn_3.48.1             XML_3.98-1.11         
[10] survival_2.42-6        rlang_0.2.1            pillar_1.2.3          
[13] glue_1.2.0             affy_1.58.0            bindrcpp_0.2.2        
[16] affyio_1.50.0          foreach_1.4.4          bindr_0.1.1           
[19] plyr_1.8.4             mzID_1.18.0            zlibbioc_1.26.0       
[22] munsell_0.5.0          pcaMethods_1.72.0      gtable_0.2.0          
[25] codetools_0.2-15       IRanges_2.14.10        doParallel_1.0.11     
[28] BiocInstaller_1.30.0   MassSpecWavelet_1.46.0 preprocessCore_1.42.0 
[31] scales_0.5.0           limma_3.36.2           S4Vectors_0.18.3      
[34] RANN_2.5.1             impute_1.54.0          ggplot2_3.0.0         
[37] digest_0.6.15          dplyr_0.7.6            grid_3.5.0            
[40] tools_3.5.0            lazyeval_0.2.1         tibble_1.4.2          
[43] pkgconfig_2.0.1        MASS_7.3-50            Matrix_1.2-14         
[46] assertthat_0.2.0       iterators_1.0.10       R6_2.2.2              
[49] MALDIquant_1.18        multtest_2.36.0        compiler_3.5.0

**Comment by jotsetung** [comment]
could you please re-run the code disabling parallel processing (i.e. after `register(SerialParam())`). Error messages from code run in parallel are always very cryptic.

**Comment by marcomtzmtz** [comment]
Thank you for your help. 
This is all the code:

library(xcms)
library(faahKO)
library(RColorBrewer)
library(pander)
library(magrittr)
register(SerialParam())

cdfs <- dir("cdfs", full.names = TRUE,
        recursive = TRUE)
pd <- data.frame(sample_name = sub(basename(cdfs), pattern = ".CDF",
                  replacement = "", fixed = TRUE),
         sample_group = c(rep("Caso", 5), rep("Control", 5)),
         stringsAsFactors = FALSE)

raw_data <- readMSData(files = cdfs, pdata = new("NAnnotatedDataFrame", pd),
               mode = "onDisk")

head(rtime(raw_data))
mzs <- mz(raw_data)

mzs_by_file <- split(mzs, f = fromFile(raw_data))
length(mzs_by_file)

bpis <- chromatogram(raw_data, aggregationFun = "max")


traceback()
18: stop(e)
17: value[[3L]](cond)
16: tryCatchOne(expr, names, parentenv, handlers[[1L]])
15: tryCatchList(expr, classes, parentenv, handlers)
14: tryCatch({
        FUN(...)
    }, error = handle_error)
13: withCallingHandlers({
        tryCatch({
            FUN(...)
        }, error = handle_error)
    }, warning = handle_warning)
12: FUN(X[[i]], ...)
11: lapply(X, FUN, ...)
10: bplapply(X = seq_along(ddd[[1L]]), wrap, .FUN = FUN, .ddd = ddd, 
        .MoreArgs = MoreArgs, BPREDO = BPREDO, BPPARAM = BPPARAM)
9: bplapply(X = seq_along(ddd[[1L]]), wrap, .FUN = FUN, .ddd = ddd, 
       .MoreArgs = MoreArgs, BPREDO = BPREDO, BPPARAM = BPPARAM)
8: bpmapply(subs_by_file, match(fileNames(subs), fns), FUN = function(cur_sample, 
       cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE)
7: bpmapply(subs_by_file, match(fileNames(subs), fns), FUN = function(cur_sample, 
       cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE)
6: withCallingHandlers(expr, warning = function(w) invokeRestart("muffleWarning"))
5: suppressWarnings(res <- bpmapply(subs_by_file, match(fileNames(subs), 
       fns), FUN = function(cur_sample, cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE))
4: .extractMultipleChromatograms(object, rt = rt, mz = mz, aggregationFun = aggregationFun, 
       missingValue = missing, msLevel = msLevel, BPPARAM = BPPARAM)
3: .local(object, ...)
2: chromatogram(raw_data, aggregationFun = "max")
1: chromatogram(raw_data, aggregationFun = "max")

> sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.5

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils    
[6] datasets  methods   base     

other attached packages:
 [1] magrittr_1.5        pander_0.6.2       
 [3] RColorBrewer_1.1-2  faahKO_1.20.0      
 [5] xcms_3.2.0          MSnbase_2.6.1      
 [7] ProtGenerics_1.12.0 mzR_2.14.0         
 [9] Rcpp_0.12.17        BiocParallel_1.14.1
[11] Biobase_2.40.0      BiocGenerics_0.26.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4       purrr_0.2.5           
 [3] splines_3.5.0          lattice_0.20-35       
 [5] colorspace_1.3-2       stats4_3.5.0          
 [7] yaml_2.1.19            vsn_3.48.1            
 [9] XML_3.98-1.11          survival_2.42-6       
[11] rlang_0.2.1            pillar_1.2.3          
[13] glue_1.2.0             affy_1.58.0           
[15] bindrcpp_0.2.2         affyio_1.50.0         
[17] foreach_1.4.4          bindr_0.1.1           
[19] plyr_1.8.4             mzID_1.18.0           
[21] zlibbioc_1.26.0        munsell_0.5.0         
[23] pcaMethods_1.72.0      gtable_0.2.0          
[25] codetools_0.2-15       IRanges_2.14.10       
[27] doParallel_1.0.11      BiocInstaller_1.30.0  
[29] MassSpecWavelet_1.46.0 preprocessCore_1.42.0 
[31] scales_0.5.0           limma_3.36.2          
[33] S4Vectors_0.18.3       RANN_2.5.1            
[35] impute_1.54.0          ggplot2_3.0.0         
[37] digest_0.6.15          dplyr_0.7.6           
[39] grid_3.5.0             tools_3.5.0           
[41] lazyeval_0.2.1         tibble_1.4.2          
[43] pkgconfig_2.0.1        MASS_7.3-50           
[45] Matrix_1.2-14          assertthat_0.2.0      
[47] iterators_1.0.10       R6_2.2.2              
[49] MALDIquant_1.18        multtest_2.36.0       
[51] compiler_3.5.0

**Comment by jotsetung** [comment]
thanks for the output from the traceback - but I can not see any helpful information there. What was the error message that you got in the call above?

Can you also please provide the output of the following calls:

```r
lapply(split(rtime(raw_data), fromFile(raw_data)), range, na.rm = TRUE)

lapply(split(mz(raw_data), fromFile(raw_data)), range, na.rm = TRUE)
```

**Comment by marcomtzmtz** [comment]
After introducing the next command, this is the error:

**bpis <- chromatogram(raw_data, aggregationFun = "max")**
Error in if (sum(sel) == 0) { : missing value where TRUE/FALSE needed

**Also, this warning appears when I read the CDF:**

> raw_data <- readMSData(files = cdfs, pdata = new("NAnnotatedDataFrame", pd),
+                mode = "onDisk")

Polarity can not be extracted from netCDF files, please set manually the polarity with the 'polarity' method.

** The next are the answer to the commands that you asked for**

lapply(split(rtime(raw_data), fromFile(raw_data)), range, na.rm = TRUE)
$`1`
[1]  205.554 3539.518

$`2`
[1]  205.547 3539.511

$`3`
[1]  205.556 3539.520

$`4`
[1]  205.556 3539.519

$`5`
[1]  205.551 3539.514

$`6`
[1]  205.554 3539.518

$`7`
[1]  205.548 3539.512

$`8`
[1]  205.556 3539.520

$`9`
[1]  205.546 3539.509

$`10`
[1]  205.551 3539.514

lapply(split(mz(raw_data), fromFile(raw_data)), range, na.rm = TRUE)
$`1`
[1]  50.1 386.3

$`2`
[1]  50.1 386.3

$`3`
[1]  50.8 386.4

$`4`
[1]  50.1 387.4

$`5`
[1]  50.1 387.4

$`6`
[1]  50.1 387.3

$`7`
[1]  50.1 387.3

$`8`
[1]  50.1 355.0

$`9`
[1]  50.1 386.3

$`10`
[1]  50.1 281.1

**Comment by jotsetung** [comment]
I added a tentative fix to `MSnbase`. Could you please install the most recent `MSnbase` from github:
`devtools::install_github("lgatto/MSnbase", ref = "fix_filterMz")` and test again?

Let me also know if you have problems installing `MSnbase`.

**Comment by marcomtzmtz** [comment]
Hello, thank you again for your time and help. 

I got the next error when I try to install MSnbase

> devtools::install_github("lgatto/MSnbase", ref = "fix_filterMz")
Downloading GitHub repo lgatto/MSnbase@fix_filterMz
from URL https://api.github.com/repos/lgatto/MSnbase/zipball/fix_filterMz
Installing MSnbase
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/hd/j8nr40jx6ws9lp20_7j3cyk40000gn/T/RtmptFunpo/devtools4cb627a2a09/lgatto-MSnbase-3b74fa6'  \
  --library='/Library/Frameworks/R.framework/Versions/3.5/Resources/library' --install-tests 

* installing *source* package ‘MSnbase’ ...
** libs
clang++ -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c RcppExports.cpp -o RcppExports.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c Spectrum1_class.c -o Spectrum1_class.o
Spectrum1_class.c:8:13: warning: unused function '_new_Spectrum1' [-Wunused-function]
static SEXP _new_Spectrum1(SEXP msLevel, SEXP peaksCount, SEXP rt,
            ^
1 warning generated.
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c Spectrum2_class.c -o Spectrum2_class.o
Spectrum2_class.c:9:13: warning: unused function '_new_Spectrum2' [-Wunused-function]
static SEXP _new_Spectrum2(SEXP msLevel, SEXP peaksCount, SEXP rt,
            ^
1 warning generated.
clang++ -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c imp_neighbour_avg.cpp -o imp_neighbour_avg.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c utils.c -o utils.o
utils.c:92:15: warning: unused variable 'ret' [-Wunused-variable]
  int i1, i2, ret;
              ^
utils.c:107:15: warning: unused variable 'ret' [-Wunused-variable]
  int i1, i2, ret;
              ^
utils.c:123:18: warning: unused variable 'ret' [-Wunused-variable]
  double d1, d2, ret;
                 ^
utils.c:133:18: warning: unused variable 'ret' [-Wunused-variable]
  double d1, d2, ret;
                 ^
utils.c:121:12: warning: unused function 'compar_double_asc_order' [-Wunused-function]
static int compar_double_asc_order(const void *p1, const void *p2)
           ^
utils.c:131:12: warning: unused function 'compar_double_desc_order' [-Wunused-function]
static int compar_double_desc_order(const void *p1, const void *p2)
           ^
6 warnings generated.
clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o MSnbase.so RcppExports.o Spectrum1_class.o Spectrum2_class.o imp_neighbour_avg.o utils.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Library/Frameworks/R.framework/Versions/3.5/Resources/library/MSnbase/libs
** R
** data
*** moving datasets to lazyload DB
Warning in fun(libname, pkgname) :
  mzR has been built against a different Rcpp version (0.12.16)
than is installed on your system (0.12.17). This might lead to errors
when loading mzR. If you encounter such issues, please send a report,
including the output of sessionInfo() to the Bioc support forum at 
https://support.bioconductor.org/. For details see also
https://github.com/sneumann/mzR/wiki/mzR-Rcpp-compiler-linker-issue.
Creating a new generic function for ‘smooth’ in package ‘MSnbase’
Creating a new generic function for ‘trimws’ in package ‘MSnbase’
in method for ‘coerce’ with signature ‘"MSnSet","SummarizedExperiment"’: no definition for class “SummarizedExperiment”
in method for ‘coerce’ with signature ‘"SummarizedExperiment","MSnSet"’: no definition for class “SummarizedExperiment”
in method for ‘coerce’ with signature ‘"IBSpectra","MSnSet"’: no definition for class “IBSpectra”
** inst
** tests
** byte-compile and prepare package for lazy loading
Error : package ‘mzR’ 2.14.0 was found, but >= 2.15.1 is required by ‘MSnbase’
ERROR: lazy loading failed for package ‘MSnbase’
* removing ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library/MSnbase’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library/MSnbase’
Installation failed: Command failed (1)

**Comment by marcomtzmtz** [comment]
I installed it (with no problem) but now the error is: 
> bpis <- chromatogram(raw_data, aggregationFun = "max")
Error in if (all(is.na(sel)) || sum(sel) == 0) { : 
  missing value where TRUE/FALSE needed
> traceback()
18: stop(e)
17: value[[3L]](cond)
16: tryCatchOne(expr, names, parentenv, handlers[[1L]])
15: tryCatchList(expr, classes, parentenv, handlers)
14: tryCatch({
        FUN(...)
    }, error = handle_error)
13: withCallingHandlers({
        tryCatch({
            FUN(...)
        }, error = handle_error)
    }, warning = handle_warning)
12: FUN(X[[i]], ...)
11: lapply(X, FUN, ...)
10: bplapply(X = seq_along(ddd[[1L]]), wrap, .FUN = FUN, .ddd = ddd, 
        .MoreArgs = MoreArgs, BPREDO = BPREDO, BPPARAM = BPPARAM)
9: bplapply(X = seq_along(ddd[[1L]]), wrap, .FUN = FUN, .ddd = ddd, 
       .MoreArgs = MoreArgs, BPREDO = BPREDO, BPPARAM = BPPARAM)
8: bpmapply(subs_by_file, match(fileNames(subs), fns), FUN = function(cur_sample, 
       cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE)
7: bpmapply(subs_by_file, match(fileNames(subs), fns), FUN = function(cur_sample, 
       cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE)
6: withCallingHandlers(expr, warning = function(w) invokeRestart("muffleWarning"))
5: suppressWarnings(res <- bpmapply(subs_by_file, match(fileNames(subs), 
       fns), FUN = function(cur_sample, cur_file, rtm, mzm, aggFun) {
       sps <- spectra(cur_sample)
       rts <- rtime(cur_sample)
       cur_res <- vector("list", nrow(rtm))
       for (i in 1:nrow(rtm)) {
           in_rt <- rts >= rtm[i, 1] & rts <= rtm[i, 2]
           if (!any(in_rt)) {
               cur_res[[i]] <- MSnbase::Chromatogram(filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun)
               next
           }
           cur_sps <- lapply(sps[in_rt], function(spct, filter_mz, 
               aggFun) {
               spct <- filterMz(spct, filter_mz)
               if (!spct@peaksCount) 
                   return(c(NA_real_, NA_real_, missingValue, NA_real_))
               c(range(spct@mz, na.rm = TRUE, finite = TRUE), do.call(aggFun, 
                   list(spct@intensity, na.rm = TRUE)), spct@msLevel)
           }, filter_mz = mzm[i, ], aggFun = aggFun)
           allVals <- unlist(cur_sps, use.names = FALSE)
           int_idx <- seq(3, length(allVals), by = 4)
           mslevel_idx <- seq(4, length(allVals), by = 4)
           ints <- allVals[int_idx]
           names(ints) <- names(cur_sps)
           mz_range <- mzm[i, ]
           if (!all(is.na(ints))) 
               mz_range <- range(allVals[-c(int_idx, mslevel_idx)], 
                   na.rm = TRUE, finite = TRUE)
           cur_res[[i]] <- MSnbase::Chromatogram(rtime = rts[in_rt], 
               intensity = ints, mz = mz_range, filterMz = mzm[i, 
                   ], fromFile = as.integer(cur_file), aggregationFun = aggFun, 
               msLevel = as.integer(unique(allVals[mslevel_idx])))
       }
       cur_res
   }, MoreArgs = list(rtm = rt, mzm = mz, aggFun = aggregationFun), 
       BPPARAM = BPPARAM, SIMPLIFY = FALSE))
4: .extractMultipleChromatograms(object, rt = rt, mz = mz, aggregationFun = aggregationFun, 
       missingValue = missing, msLevel = msLevel, BPPARAM = BPPARAM)
3: .local(object, ...)
2: chromatogram(raw_data, aggregationFun = "max")
1: chromatogram(raw_data, aggregationFun = "max")
> sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.5

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_1.5        pander_0.6.2        RColorBrewer_1.1-2  faahKO_1.20.0      
 [5] xcms_3.2.0          MSnbase_2.6.2       ProtGenerics_1.12.0 mzR_2.14.0         
 [9] Rcpp_0.12.17        BiocParallel_1.14.2 Biobase_2.40.0      BiocGenerics_0.26.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4       purrr_0.2.5            splines_3.5.0         
 [4] lattice_0.20-35        colorspace_1.3-2       stats4_3.5.0          
 [7] yaml_2.1.19            vsn_3.48.1             XML_3.98-1.12         
[10] survival_2.42-6        rlang_0.2.1            pillar_1.3.0          
[13] glue_1.2.0             affy_1.58.0            bindrcpp_0.2.2        
[16] affyio_1.50.0          foreach_1.4.4          bindr_0.1.1           
[19] plyr_1.8.4             mzID_1.18.0            zlibbioc_1.26.0       
[22] munsell_0.5.0          pcaMethods_1.72.0      gtable_0.2.0          
[25] codetools_0.2-15       IRanges_2.14.10        doParallel_1.0.11     
[28] BiocInstaller_1.30.0   MassSpecWavelet_1.46.0 preprocessCore_1.42.0 
[31] scales_0.5.0           limma_3.36.2           S4Vectors_0.18.3      
[34] RANN_2.6               impute_1.54.0          ggplot2_3.0.0         
[37] digest_0.6.15          dplyr_0.7.6            grid_3.5.0            
[40] tools_3.5.0            lazyeval_0.2.1         tibble_1.4.2          
[43] crayon_1.3.4           pkgconfig_2.0.1        MASS_7.3-50           
[46] Matrix_1.2-14          assertthat_0.2.0       rstudioapi_0.7        
[49] iterators_1.0.10       R6_2.2.2               MALDIquant_1.18       
[52] multtest_2.36.0        compiler_3.5.0

**Comment by marcomtzmtz** [comment]
This link has my code and my data, I think it could be useful:

https://drive.google.com/open?id=1DKWpm6

Thank you for all your efforts.

**Comment by jotsetung** [comment]
Thanks for the files! Now I was able to fix the problem for good (in `MSnbase`) - only, you'll have to wait for the pull request to be accepted and the update being pushed to Bioconductor.

**Comment by marcomtzmtz** [comment]
I wil be aware of the changes. Thankyou for your help.I really appreciate it.

**Comment by jotsetung** [comment]
Could you then please close the issue once you've verified that the fix works for you?

**Comment by marcomtzmtz** [comment]
The error continues, How can I correct it?

**Comment by jotsetung** [comment]
I've tried just now and with `MSnbase` version 2.7.4 (which is the current developmental version) it works. In `MSnbase` 2.6.2 (current release version) it is not fixed. I asked @lgatto for inclusion of the fix into the release version of `MSnbase`.

**Comment by lgatto** [comment]
Following up [here](https://github.com/lgatto/MSnbase/pull/355#issuecomment-415658371).

**Comment by marcomtzmtz** [comment]
Thank you for your help. It worked, I am using version 2.7.4 (developmental). Now I have another similar error when I try to align. Do I need to close the issue and open a new? This is the error:

cwp <- CentWaveParam(ppm=100, snthresh = 6, peakwidth = c(5, 10), mzdiff = 0.01, prefilter = c(3,100), noise=0)
xdata <- findChromPeaks(raw_data, param = cwp) 
xdata <- adjustRtime(xdata, param = ObiwarpParam(binSize = 1))

The error is the next:

Sample number 2 used as center sample.
Error in binYonX(mz, int, breaks = brks, fromIdx = fromIdx, toIdx = toIdx,  : 
  No 'NA' values are allowed in 'x'!

Let me know if I need to open a new issue, but I think it has to deal with NAs and a similar function.

##traceback()

25: stop(e)
24: value[[3L]](cond)
23: tryCatchOne(expr, names, parentenv, handlers[[1L]])
22: tryCatchList(expr, classes, parentenv, handlers)
21: tryCatch({
        FUN(...)
    }, error = handle_error)
20: withCallingHandlers({
        tryCatch({
            FUN(...)
        }, error = handle_error)
    }, warning = handle_warning)
19: FUN(...)
18: FUN(X[[i]], ...)
17: lapply(X, FUN_, ...)
16: bplapply(X, FUN, ..., BPREDO = BPREDO, BPPARAM = BPPARAM)
15: bplapply(X, FUN, ..., BPREDO = BPREDO, BPPARAM = BPPARAM)
14: bplapply(splitByFile(object, f = theF), function(z, bmethod, 
        bstep, bbaselevel, bbasespace, bmzrange., breturnBreaks) {
        require(xcms, quietly = TRUE)
        sps <- spectra(z, BPPARAM = SerialParam())
        mzs <- lapply(sps, mz)
        vps <- lengths(mzs, use.names = FALSE)
        return(.createProfileMatrix(mz = unlist(mzs, use.names = FALSE), 
            int = unlist(lapply(sps, intensity), use.names = FALSE), 
            valsPerSpect = vps, method = bmethod, step = bstep, baselevel = bbaselevel, 
            basespace = bbasespace, mzrange. = bmzrange., returnBreaks = breturnBreaks))
    }, bmethod = method, bstep = step, bbaselevel = baselevel, bbasespace = basespace, 
        bmzrange. = mzrange., breturnBreaks = returnBreaks)
13: bplapply(splitByFile(object, f = theF), function(z, bmethod, 
        bstep, bbaselevel, bbasespace, bmzrange., breturnBreaks) {
        require(xcms, quietly = TRUE)
        sps <- spectra(z, BPPARAM = SerialParam())
        mzs <- lapply(sps, mz)
        vps <- lengths(mzs, use.names = FALSE)
        return(.createProfileMatrix(mz = unlist(mzs, use.names = FALSE), 
            int = unlist(lapply(sps, intensity), use.names = FALSE), 
            valsPerSpect = vps, method = bmethod, step = bstep, baselevel = bbaselevel, 
            basespace = bbasespace, mzrange. = bmzrange., returnBreaks = breturnBreaks))
    }, bmethod = method, bstep = step, bbaselevel = baselevel, bbasespace = basespace, 
        bmzrange. = mzrange., breturnBreaks = returnBreaks)
12: .local(object, ...)
11: profMat(object, method = "bin", step = binSize(param), fileIndex = centerSample(param), 
        returnBreaks = TRUE)
10: profMat(object, method = "bin", step = binSize(param), fileIndex = centerSample(param), 
        returnBreaks = TRUE)
9: withCallingHandlers(expr, message = function(c) invokeRestart("muffleMessage"))
8: suppressMessages(profCtr <- profMat(object, method = "bin", step = binSize(param), 
       fileIndex = centerSample(param), returnBreaks = TRUE)[[1]])
7: xcms:::.obiwarp(object_sub, param = param)
6: .local(object, param, ...)
5: adjustRtime(as(object, "OnDiskMSnExp"), param = param, msLevel = msLevel)
4: adjustRtime(as(object, "OnDiskMSnExp"), param = param, msLevel = msLevel)
3: .local(object, param, ...)
2: adjustRtime(xdata, param = ObiwarpParam(binSize = 1))
1: adjustRtime(xdata, param = ObiwarpParam(binSize = 1))

##sessionInfo

R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.5

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices
[5] utils     datasets  methods   base     

other attached packages:
 [1] magrittr_1.5        pander_0.6.2       
 [3] RColorBrewer_1.1-2  faahKO_1.21.0      
 [5] xcms_3.3.2          MSnbase_2.7.4      
 [7] ProtGenerics_1.13.0 mzR_2.15.1         
 [9] Rcpp_0.12.18        BiocParallel_1.15.9
[11] Biobase_2.41.2      BiocGenerics_0.27.1

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4      
 [2] purrr_0.2.5           
 [3] splines_3.5.1         
 [4] lattice_0.20-35       
 [5] colorspace_1.3-2      
 [6] stats4_3.5.1          
 [7] yaml_2.2.0            
 [8] vsn_3.49.1            
 [9] XML_3.98-1.16         
[10] survival_2.42-6       
[11] rlang_0.2.2           
[12] pillar_1.3.0          
[13] glue_1.3.0            
[14] affy_1.59.0           
[15] bindrcpp_0.2.2        
[16] affyio_1.51.0         
[17] foreach_1.4.4         
[18] bindr_0.1.1           
[19] plyr_1.8.4            
[20] mzID_1.19.0           
[21] robustbase_0.93-2     
[22] zlibbioc_1.27.0       
[23] munsell_0.5.0         
[24] pcaMethods_1.73.0     
[25] gtable_0.2.0          
[26] codetools_0.2-15      
[27] IRanges_2.15.17       
[28] doParallel_1.0.11     
[29] BiocInstaller_1.31.3  
[30] MassSpecWavelet_1.47.0
[31] preprocessCore_1.43.0 
[32] DEoptimR_1.0-8        
[33] scales_1.0.0          
[34] limma_3.37.3          
[35] S4Vectors_0.19.19     
[36] RANN_2.6              
[37] impute_1.55.0         
[38] ggplot2_3.0.0         
[39] digest_0.6.16         
[40] dplyr_0.7.6           
[41] grid_3.5.1            
[42] tools_3.5.1           
[43] lazyeval_0.2.1        
[44] tibble_1.4.2          
[45] crayon_1.3.4          
[46] pkgconfig_2.0.2       
[47] MASS_7.3-50           
[48] Matrix_1.2-14         
[49] assertthat_0.2.0      
[50] rstudioapi_0.7        
[51] iterators_1.0.10      
[52] R6_2.2.2              
[53] MALDIquant_1.18       
[54] multtest_2.37.0       
[55] compiler_3.5.1

**Comment by jotsetung** [comment]
Please open another issue on your new problem (and close this one as it seems the original problem is fixed). Is it based on the same files? than I could check what happens.

**Comment by marcomtzmtz** [comment]
Yes is the same data. Thank you. The error of extracting chromatogram was corrected.

**Comment by SONGKai-PKU** [comment]
Dear xcms devel:
  I get the following error when trying to read netCDF files (actually GC×GC chromatograms which could be read in the CDF form):

> cooking_raw_data <- readMSData(files = list.files(), pdata = new("NAnnotatedDataFrame", cooking_meta), mode = "onDisk") 
Polarity can not be extracted from netCDF files, please set manually the polarity with the 'polarity' method.
[1] "vobjtovarid4: error #F: I could not find the requsted var (or dimvar) in the file!"
[1] "var (or dimvar) name: instrument_name"
[1] "file name: F:\\TDGCMS\\Mobile_source\\data_processing\\comprehensive_pixel\\Image_CDF_XCMS\\cooking\\CDF\\cooking_gas_01_Run01_Img01.cdf"
Error in vobjtovarid4(nc, varid, verbose = verbose, allowdimvar = TRUE) : 
  Variable not found

Thanks a lot for your time and help~
